lib/logstash/inputs/jms.rb

```diff
@@ -0,0 +1,285 @@
+# encoding: utf-8
+require "logstash/inputs/base"
+require "logstash/inputs/threadable"
+require "logstash/namespace"
+
+# Read events from a Jms Broker. Supports both Jms Queues and Topics.
+#
+# For more information about Jms, see <http://docs.oracle.com/javaee/6/tutorial/doc/bncdq.html>
+# For more information about the Ruby Gem used, see <http://github.com/reidmorrison/jruby-jms>
+# Here is a config example :
+#	 jms {
+#			include_header => false
+#			include_properties => false
+#			include_body => true
+#			use_jms_timestamp => false
+#			interval => 10
+#			queue_name => "myqueue"
+#			yaml_file => "~/jms.yml"
+#			yaml_section => "mybroker"
+#		}
+#
+#
+class LogStash::Inputs::Jms < LogStash::Inputs::Threadable
+	config_name "jms"
+
+	default :config, "plain"
+
+	# A JMS message has three parts :
+	#	 Message Headers (required)
+	#	 Message Properties (optional)
+	#	 Message Bodies (optional)
+	# You can tell the input plugin which parts should be included in the event produced by Logstash
+	#
+	# Include JMS Message Header Field values in the event
+	config :include_header, :validate => :boolean, :default => true
+	# Include JMS Message Properties Field values in the event
+	config :include_properties, :validate => :boolean, :default => true
+	# Include JMS Message Body in the event
+	# Supports TextMessage and MapMessage
+	# If the JMS Message is a TextMessage, then the value will be in the "message" field of the event
+	# If the JMS Message is a MapMessage, then all the key/value pairs will be added in the Hashmap of the event
+	# BytesMessage, StreamMessage and ObjectMessage are not supported
+	config :include_body, :validate => :boolean, :default => true
+	# Convert the JMSTimestamp header field to the @timestamp value of the event
+	# Don't use it for now, it is buggy
+	config :use_jms_timestamp, :validate => :boolean, :default => false
+
+	# Choose an implementation of the run block. Value can be either consumer, async or thread
+	config :runner, :validate => :string, :required => true, :default => "consumer"
```

Comment:
here you can validate the 3 options using

`config :runner, :validate => [ "consumer", "async", "thread" ], :default => "consumer"`

also don't need the required, since you've specified a default